src/commonMain/kotlin/net/kyori/adventure/webui/websocket/Packet.kt

```diff
@@ -0,0 +1,34 @@
+package net.kyori.adventure.webui.websocket
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonObject
+
+/** The client -> server call. */
+public sealed interface Packet
+
+public sealed interface Templates : Packet {
+    public val stringTemplates: Map<String, String>?
+    public val componentTemplates: Map<String, JsonObject>?
+    public val miniMessageTemplates: Map<String, String>?
+}
+
+@Serializable
+@SerialName("call")
+public data class Call(public val miniMessage: String? = null) : Packet
+
+@Serializable
+@SerialName("templates")
+public data class TemplatesImpl(
+    public override val stringTemplates: Map<String, String>? = null,
+    public override val componentTemplates: Map<String, JsonObject>? = null,
+    public override val miniMessageTemplates: Map<String, String>? = null
+) : Templates
+
+@Serializable
+public data class Combined(
```

Comment:
I think it would be nicer (and would also avoid the need for the defined polywhatever serialisation registration module thing, if this data class just contained a call and templates class, rather than implementing templates. Then we wouldn't need the templates interface and the `when` statement later on could be exhaustive as well.

Reply:
I tried to structure it to be nice in JSON structure, but I sorta see what you mean! I will redo these data classes